config/config.go

```diff
@@ -85,6 +86,9 @@ type Config struct {
 	// generation and dependency resolution.
 	Repos []*rule.Rule
 
+	// go_repository rules, mapped by name.
+	GoRepos map[string]repo.Repo
```

Comment:
This is where it would have been very nice to have the repo cache be Go specific instead of pretending to be generic. I just can't get information across to the resolver because it passes through a "generic" conduit.

I can do simply use the repos, but I think it's a linear pass through all the repo rules every time we want to resolve an external dep. That _sounds_ like a lot to me, we will end up having hundreds of repo rules and many thousand deps.

Any ideas?